File: data/multilang_comparison/program_cpp_array/src/func24.cpp
#include "program_cpp_array.hpp" 
Array* func24(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array_param params0;
      params0.size = 0;
      Array* data_params0[0];
      params0.data = data_params0;
      Array* array68 = func25(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array68->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array68->refC--;
      if(array68->refC == 0) {
         delete[] array68->data;
         array68->data = nullptr;
         DEBUG_FREE(array68->id);
         delete array68;
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      Array* data_params0[0];
      params0.data = data_params0;
      Array* array77 = func26(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array77->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array77->refC--;
      if(array77->refC == 0) {
         delete[] array77->data;
         array77->data = nullptr;
         DEBUG_FREE(array77->id);
         delete array77;
      }
   }
   Array_param params0;
   params0.size = 0;
   Array* data_params0[0];
   params0.data = data_params0;
   Array* array82 = func27(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array82->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   return array82;
}

File: data/multilang_comparison/program_go/program_go/src/func24.go
package main
import "math/rand"
var _ = rand.Intn
func Func24(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array99 *Array = Func25(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array99.RefC--
      if array99.RefC == 0 {
          array99.Data = nil
          array99 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array109 *Array = Func26(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array109.RefC--
      if array109.RefC == 0 {
          array109.Data = nil
          array109 = nil
      }
   }
   var params0 ArrayParam
   params0.Size = 0
   params0.Data = make([]*Array, params0.Size)
   var array114 *Array = Func27(&params0, getPath(), loopsFactor);
   params0.Data = nil
   params0.Size = 0
   return array114;
}

